### PR TITLE
feat(CLI): #27416 Adding GHA release CLI workflow steps for publishing binaries into NPM registry.

### LINKED ISSUE #27416: CLI: Publishing dotcms-cli distributions on NPM registry
Publishing `dotcms-cli` distributions on the NPM (Node Package Manager) registry involves making the dotCMS command-line interface (CLI) packages available for installation and use by developers and users within the `Node.js` ecosystem. It is necessary to integrate into the current dotcms-cli release workflow following the steps described in #27089

### PR DESCRIPTION
**Pull request by dcolina**
### Proposed Changes
The `publish-npm-package` job in this GitHub workflow is responsible for publishing an npm package to the public npm registry. Here's a summary of what this job does:

**Environment Setup:**

Sets up the `Node.js `environment using the actions/setup-node action.
Installs the `jinja2-cli` tool using pip.

**Generating Package Version Suffix:**

Determines if the package version is a release version or a snapshot version.
If it's a snapshot version, it checks if a release candidate version already exists and generates the next release candidate version number.
Sets the `NPM_PACKAGE_VERSION` environment variable with the npm package version to be used and `NPM_PACKAGE_VERSION_TAG` with the npm package tag.

**Preparing the npm Package:**

Creates a folder structure for the npm package and prepares necessary files, including the `package.json`, `README.md`, and a `post-installation` script.

**Publishing to npm Registry:**

Sets up the npm authentication token for authentication.
Publishes the npm package to the public npm registry using the npm publish command, specifying public access and the appropriate tag (`latest` for release versions, `rc` for release candidate versions).
In summary, this job automates the process of publishing an npm package to the public npm registry, handling both release versions and snapshot versions and release candidates.

On the other hand, a wrapper script is added to install and uninstall a command-line interface (CLI) tool by creating or removing symbolic links to the relevant binary executable files. 

### Fixes
#27416

### CODE REVIEW

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

File: .github/workflows/cli-release-process.yml [hunk 1]
```diff
@@ -19,7 +19,10 @@ env:
   JAVA_DISTRO: temurin
   NEXT_VERSION: '1.0.0-SNAPSHOT'
   GRAALVM_VERSION: '22.1.0'
-  PACKAGE_TYPE: 'uber-jar'
+  MVN_PACKAGE_TYPE: 'uber-jar'
+  NPM_PACKAGE_NAME: 'dotcli-test'
```
Comment (fmontes) [inline .github/workflows/cli-release-process.yml:23]:
```suggestion
  NPM_PACKAGE_NAME: 'dotcli'
```

Are we sure we want to call it `test`?

**Review by fmontes — APPROVED** [review]

File: .github/workflows/cli-release-process.yml (hunk 1 above)
Comment (dcolina) [reply .github/workflows/cli-release-process.yml:23]: You are right, it's a mistake.
